### DISCUSSION TITLE
Keycloak cannot find Driver Class for DB Driver

### DISCUSSION THREAD

**Discussion by dennisn00** [keycloak/keycloak#21315] [Q&A]
Hi everyone, 
I am trying to connect Keycloak (running as Docker container) to my CrateDB instance. Unfortunately, the standard Postgres Driver does not work since Crate doesn't offer full SQL compatibility. So instead, I was trying to connect using the Crate JDBC Driver. In the dockerfile, I am downloading the Driver jar and then add it to the Classpath. My Dockerfile looks like this.
```
FROM quay.io/keycloak/keycloak:latest as builder
ENV KC_DB=postgres

WORKDIR /opt/keycloak

RUN keytool -genkeypair -storepass password -storetype PKCS12 -keyalg RSA -keysize 2048 -dname "CN=server" -alias server -ext "SAN:c=DNS:localhost,IP:127.0.0.1" -keystore conf/server.keystore
RUN /opt/keycloak/bin/kc.sh build

FROM quay.io/keycloak/keycloak:latest
COPY --from=builder /opt/keycloak/ /opt/keycloak/

ADD --chown=keycloak:keycloak https://repo1.maven.org/maven2/io/crate/crate-jdbc-standalone/2.7.0/crate-jdbc-standalone-2.7.0.jar /opt/keycloak/lib/lib/main/

ENV CLASSPATH=/opt/keycloak/lib/lib/main/crate-jdbc-standalone-2.7.0.jar

ENV KC_DB=postgres
ENV KC_DB_URL=jdbc:postgresql://<ip_address>:5432/doc
ENV KC_DB_USERNAME=user
ENV KC_DB_PASSWORD=password
ENV KC_DB_DRIVER=io.crate.client.jdbc.CrateDriver
ENTRYPOINT ["/opt/keycloak/bin/kc.sh"]
```

Unfortunately, I get an Exception:
`
java.lang.ClassNotFoundException: io.crate.client.jdbc.CrateDriver
`
I can see that the jar is downloaded to the specifies directory and I am confident that the Driver name is correct. 
What else could be the problem for keycloak not being able to find the CrateDriver?

**Comment by weltonrodrigo** [comment]
I'm not sure Keycloak support other databases beyond mariadb, mssql, mysql, oracle and postgres, but if you really want to, you can try:

- running `RUN /opt/keycloak/bin/kc.sh build` after setting `KC_DB_`*  options. **OR**
- downloading the jdbc driver to /opt/keycloak/providers
